src/main/java/org/hydev/mcpm/server/crawlers/spiget/CreateDatabase.java

```diff
@@ -0,0 +1,174 @@
+import org.hydev.mcpm.client.models.PluginModel;
+import org.hydev.mcpm.client.models.PluginVersion;
+import org.hydev.mcpm.client.models.PluginYml;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+import org.yaml.snakeyaml.error.YAMLException;
+
+import java.io.BufferedInputStream;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+
+/**
+ * Responsible for statically generating database "index" files for mcpm.
+ * Used for searches, etc. To create a database use the createDatabase method
+ * with a path to the .mcpm/pkgs/spiget directory generated previously by SpigetCrawler.
+ */
+public class CreateDatabase {
+    public static final String packageStore = ".mcpm";
+
+    /**
+     * Main method generates a database in the mcpm directory.
+     *
+     * @param args Arguments are ignored.
+     */
+    public static void main(String[] args) {
+        writeDatabase(new File(packageStore, "pkgs/spiget"), new File(packageStore, "db"));
+    }
+
+    /**
+     * Iterates through the crawlerDirectory and assembles and writes a database object to databaseFile.
+     *
+     * @param crawlerDirectory Path to the .mcpm/pkgs/spiget directory generated by SpigetCrawler.
+     * @param databaseFile Path to the db.json file that will be written to.
+     */
+    public static void writeDatabase(File crawlerDirectory, File databaseFile) {
+        Database database = createDatabase(crawlerDirectory);
+
+        if (database == null)
+            return;
+
+        try {
+            new ObjectMapper().writeValue(databaseFile, database);
+        } catch (IOException e) {
+            e.printStackTrace();
+
+            System.out.println("Failed to write database file.");
+        }
+    }
+
+    /**
+     * Returns a Database object created from the contents of the crawler directory.
+     *
+     * @param crawlerDirectory The directory generated by SpigetCrawler that contains package info.
+     * @return A database object containing information about the valid plugins.
+     */
+    @Nullable
+    public static Database createDatabase(File crawlerDirectory) {
+        var files = crawlerDirectory.listFiles();
+
+        if (files == null) {
+            System.err.println("Missing contents of directory");
+
+            return null;
+        }
+
+        List<PluginModel> plugins = Arrays.stream(files)
+            .map(CreateDatabase::createPluginModel)
+            .filter(Optional::isPresent)
+            .map(Optional::get)
+            .toList();
+
+        return new Database(plugins);
+    }
+
+    @NotNull
+    private static Optional<PluginModel> createPluginModel(File directory) {
+        try {
+            var id = Long.parseLong(directory.getName());
```

Comment:
I see the exception being caught but are directory names just integers

Reply:
The spiget API that Azalea writes outputs directories with names corresponding to their plugin id (which are just integers, yep). We want to be prudent, so when we're fetching the plugin id we want to be able to handle cases where some directory is a rebel and has a non-integer name (we just skip it, it's probably not a plugin).